### PR TITLE
Update macOS page, remove old examples

### PR DESCRIPTION
**Pull request by devinmarieb**
# Description

- Remove examples using unsupported images.
- Minor updates to content.

# Reasons
Two images were being referenced that are no longer being maintained which had already been removed from another page.
[Jira ticket](https://circleci.atlassian.net/jira/software/c/projects/DOCTEAM/boards/554?modal=detail&selectedIssue=DOCTEAM-421)

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [x] Break up walls of text by adding paragraph breaks.
- [x] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [x] Keep the title between 20 and 70 characters.
- [x] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [x] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [x] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [x] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/using-macos.md [hunk 1]
```diff
@@ -263,19 +248,19 @@ React Native projects can be built on CircleCI using `macos` and `docker` execut
 ### Creating a `config.yml` File
 {: #creating-a-configyml-file }
 
-The most flexible way to customize your build is to modify the CircleCI configuration for your project in `.circleci/config.yml`. This allows you to run arbitrary bash commands as well as use built-in features such as workspaces and caching. See the [Configuring CircleCI]( {{ site.baseurl }}/configuration-reference/) documentation for a detailed description of the structure of the `.circleci/config.yml` file.
+The most flexible way to customize your build is to modify the CircleCI configuration for your project in `.circleci/config.yml`. This allows you to run arbitrary bash commands as well as use built-in features such as [workspaces]({{site.baseurl}}/workspaces) and [caching]({{site.baseurl}}/caching). See the [Configuring CircleCI]({{site.baseurl}}/configuration-reference/) documentation for a detailed description of the structure of the `.circleci/config.yml` file.
 
 ## Using Multiple Executor Types (macOS + Docker)
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-macos.md:253]: @devinmarieb I think what you have done with this example is totally fine. IMO we could just remove this whole section though and we can add more generic information on using multiple executors in a single pipeline or workflow to the Intro to Execution Envs doc. What do you think? We have this example we could use: /docs/sample-config#sample-configuration-with-multiple-executor-types ... There are some other things we've talked about doing in that intro doc so I'm happy to make a ticket

File: jekyll/_cci2/using-macos.md [hunk 2]
```diff
@@ -263,19 +248,19 @@ React Native projects can be built on CircleCI using `macos` and `docker` execut
 ### Creating a `config.yml` File
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-macos.md:248]: I feel like this section isn't really needed either?

File: jekyll/_cci2/using-macos.md [hunk 3]
```diff
@@ -92,55 +93,37 @@ Running or testing Apple Silicon apps natively is currently not possible as Circ
...
-command:
+Assuming that you want to create a standalone `x86_64` binary from a universal binary called `circleci-demo-macos`, you can do so by running the command:
 
 ```shell
 lipo -extract x86_64 circleci-demo-macos.app/Contents/MacOS/circleci-demo-macos -output circleci-demo-macos-x86_64
 ```
 
-You can then confirm the supported architecture of the extracted binary with
-`lipo -info circleci-demo-macos-x86_64`, which will output the following
+You can then confirm the supported architecture of the extracted binary with `lipo -info circleci-demo-macos-x86_64`, which will output the following:
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-macos.md:109]: `lipo -info circleci-demo-macos-x86_64` should be in it's own code block, and I think using `shell` formatting is correct for all three items here - two runnable commands and one example output in the terminal.

File: jekyll/_cci2/using-macos.md (hunk 1 above)
Comment (devinmarieb) [reply jekyll/_cci2/using-macos.md:253]: @rosieyohannan Yeah that could be a solution. The second example uses community driven Docker images (from Docker community not ours) and while they have hundreds of thousands of downloads, they haven't been updated in years (one of them over four years). Another convo around using images/orbs in examples that are community driven and not officially maintained I guess.

**Comment by devinmarieb** [comment]
@rosieyohannan I made some changes but kept the example in there as it's a much less convoluted example than the one on the /sample-config page. As I mentioned, that example uses a bunch of third party community driven things which can be confusing if you're just looking for a basic example.

File: jekyll/_cci2/using-macos.md [hunk 4]
```diff
@@ -260,22 +251,17 @@ The following common test tools are known to work well on CircleCI:
...
-### Creating a `config.yml` File
-{: #creating-a-configyml-file }
-
-The most flexible way to customize your build is to modify the CircleCI configuration for your project in `.circleci/config.yml`. This allows you to run arbitrary bash commands as well as use built-in features such as workspaces and caching. See the [Configuring CircleCI]( {{ site.baseurl }}/configuration-reference/) documentation for a detailed description of the structure of the `.circleci/config.yml` file.
-
 ## Using Multiple Executor Types (macOS + Docker)
 {: #using-multiple-executor-types-macos-docker }
 
-It is possible to use multiple [executor types]({{site.baseurl}}/executor-intro/) in the same workflow. In the following example each push of an iOS project will be built on macOS, and additional iOS tools ([SwiftLint](https://github.com/realm/SwiftLint) and [Danger](https://github.com/danger/danger)) will be run in Docker.
+It is possible to use multiple [executor types]({{site.baseurl}}/executor-intro/) in the same workflow. In the following example each push of an iOS project will be built on macOS, and and a deploy image will run in Docker.
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-macos.md:257]:
```suggestion
It is possible to use multiple [executor types]({{site.baseurl}}/executor-intro/) in the same workflow. In the following example each push of an iOS project will be built on macOS, and a deploy image will run in Docker.
```

**Comment by rosieyohannan** [comment]
One typo then good to go! @devinmarieb

**Review by rosieyohannan — APPROVED** [review]

**Comment by rosieyohannan** [comment]
Approved so I don't hold you up

**Pull request merged**